src/main/java/com/onarandombox/MultiversePortals/PortalPlayerSession.java

```diff
@@ -282,31 +282,55 @@ public void setTeleportTime(Date date) {
         this.lastTeleportTime = date;
     }
 
-    public boolean allowTeleportViaCooldown(Date date) {
-        return (date.after(new Date(this.lastTeleportTime.getTime() + this.plugin.getCooldownTime())));
+    /**
+     * Get the remaining teleport cooldown time in milliseconds.
+     * If the cooldown is finished, 0 will be returned.
+     *
+     * @return The remaining teleport cooldown time in milliseconds, or 0 if the cooldown is over.
+     */
+    private long getRemainingTeleportCooldown(Date time) {
+        long cooldownEndMs = this.lastTeleportTime.getTime() + this.plugin.getCooldownTime();
+        long timeMs = time.getTime();
+        if (cooldownEndMs > timeMs) {
+            return cooldownEndMs - timeMs;
+        } else {
+            return 0;
+        }
```

Comment:
Dont really need all the these if...else, can just `return cooldownEndMs - timeMs;` Bcu you are checking that later here already:

https://github.com/Multiverse/Multiverse-Portals/blob/fbca285898058856ddad50e9e60c150ff425db7d/src/main/java/com/onarandombox/MultiversePortals/PortalPlayerSession.java#L329-L330

Reply:
Yeah, I don't see why not.